src/Features/CSharp/Portable/IntroduceVariable/CSharpIntroduceParameterService.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Diagnostics.CodeAnalysis;
+using System.Threading;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CSharp.CodeGeneration;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.IntroduceVariable;
+using Microsoft.CodeAnalysis.Simplification;
+
+namespace Microsoft.CodeAnalysis.CSharp.IntroduceVariable
+{
+    [ExportCodeRefactoringProvider(LanguageNames.CSharp), Shared]
+    internal partial class CSharpIntroduceParameterService : AbstractIntroduceParameterService<
+        CSharpIntroduceParameterService,
+        ExpressionSyntax,
+        InvocationExpressionSyntax,
+        IdentifierNameSyntax>
+    {
+        [ImportingConstructor]
+        [SuppressMessage("RoslynDiagnosticsReliability", "RS0033:Importing constructor should be [Obsolete]", Justification = "Used in test code: https://github.com/dotnet/roslyn/issues/42814")]
+        public CSharpIntroduceParameterService()
+        {
+        }
+
+        protected override SeparatedSyntaxList<SyntaxNode> AddArgumentToArgumentList(SeparatedSyntaxList<SyntaxNode> invocationArguments,
+                                                                                     SyntaxNode newArgumentExpression,
+                                                                                     int insertionIndex,
+                                                                                     string name,
+                                                                                     bool named)
+        {
+            ArgumentSyntax argument;
+            if (named)
+            {
+                var nameColon = SyntaxFactory.NameColon(name);
+                argument = SyntaxFactory.Argument(nameColon, default, (ExpressionSyntax)newArgumentExpression.WithAdditionalAnnotations(Simplifier.Annotation));
+            }
+            else
+            {
+                argument = SyntaxFactory.Argument((ExpressionSyntax)newArgumentExpression.WithAdditionalAnnotations(Simplifier.Annotation));
```

Comment:
can use SyntaxGenerator.Argument to generate the argumentyntax.  this would allow you to pull this up to base type.

Reply:
Does the SyntaxGenerator handle wanting an argument to have namecolons?

Reply:
yes.  there are overloads that can be given a name, as well as ref-kind.

Reply:
👍 pulled up